docs/source/reference-lowlevel.rst

```diff
@@ -460,7 +460,7 @@ this does serve to illustrate the basic structure of the
            self._held = False
 
        async def acquire(self):
-           if self._held:
+           while self._held:
                task = trio.lowlevel.current_task()
                self._blocked_tasks.append(task)
                def abort_fn(_):
```

Comment:
I think there's a different bug in the code then.

If this loop can run more than once, and without abort being called, then the task will be appended to `_blocked_tasks` multiple times.

This avoids the bug and makes it clear what things aren't changing on each iteration:
```python
async def acquire(self):
    if self._held:
        task = trio.lowlevel.current_task()
        self._blocked_tasks.append(task)
        def abort_fn(_):
            self._blocked_tasks.remove(task)
            return trio.lowlevel.Abort.SUCCEEDED
        # may need multiple attempts since reschedule() is not instant
        while self._held:
            await trio.lowlevel.wait_task_rescheduled(abort_fn)
    self._held = True
```